contracts/vaults/VaultPauser/RibbonVaultPauser.sol

```diff
@@ -0,0 +1,253 @@
+        address indexed vaultAddress,
+        uint256 share,
+        uint256 round
+    );
+
+    event Resume(
+        address indexed account,
+        address indexed vaultAddress,
+        uint256 withdrawAmount
+    );
+
+    event ProcessWithdrawal(address indexed vaultAddress, uint256 round);
+
+    /************************************************
+     *  CONSTRUCTOR & INITIALIZATION
+     ***********************************************/
+
+    /**
+     * @notice Initializes the contract with immutable variables
+     */
+    constructor(address _weth, address _steth) {
+        require(_weth != address(0), "!_weth");
+
+        WETH = _weth;
+        STETH = _steth;
+    }
+
+    // /**
+    //  * @notice Initializes the contract with storage variables.
+    //  */
+    function initialize(address _owner, address _keeper) external initializer {
+        __ReentrancyGuard_init();
+        __Ownable_init();
+        transferOwnership(_owner);
+        keeper = _keeper;
+    }
+
+    /**
+     * @dev Throws if called by any account other than the keeper.
+     */
+    modifier onlyKeeper() {
+        require(msg.sender == keeper, "!keeper");
+        _;
+    }
+
+    /************************************************
+     *  GETTERS
+     ***********************************************/
+
+    /**
+     * @notice gets pause position for specific vault and user
+     */
+    function getPausePositions(address _vaultAddress, address _userAddress)
+        external
+        view
+        returns (PauseReceipt[] memory)
+    {
+        return pausedPositions[_vaultAddress][_userAddress];
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice Sets the new keeper
+     * @param newKeeper is the address of the new keeper
+     */
+    function setNewKeeper(address newKeeper) external onlyOwner {
+        require(newKeeper != address(0), "!newKeeper");
+        keeper = newKeeper;
+    }
+
+    /************************************************
+     *  VAULT OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice pause position from vault by redeem all the shares from vault to Pauser
+     * @param _account user's address
+     * @param _amount the amount of shares
+     */
+    function pausePosition(address _account, uint256 _amount)
+        external
+        override
+    {
+        address currentVaultAddress = address(msg.sender);
+        IRibbonThetaVault currentVault = IRibbonThetaVault(currentVaultAddress);
+
+        PauseReceipt[] storage pauseReceipts =
+            pausedPositions[currentVaultAddress][_account];
+
+        // user can pause multiple position hence it's necessary to make an array
+        // to store each paused position
+        pauseReceipts.push(
+            PauseReceipt({
+                round: uint16(currentVault.vaultState().round),
```

Comment:
Same for `currentVault.vaultState().round`

https://github.com/ribbon-finance/ribbon-v2/pull/254/files#r884435142